server/src/test/java/io/crate/integrationtests/PgTunnelLogicalReplicationITest.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.integrationtests;
+
+import static org.hamcrest.CoreMatchers.is;
+
+import java.net.InetSocketAddress;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nullable;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.discovery.SettingsBasedSeedHostsProvider;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.InternalTestCluster;
+import org.elasticsearch.test.MockHttpTransport;
+import org.elasticsearch.test.NodeConfigurationSource;
+import org.elasticsearch.test.junit.annotations.TestLogging;
+import org.elasticsearch.transport.Netty4Plugin;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Test;
+
+import io.crate.protocols.postgres.PGErrorStatus;
+import io.crate.protocols.postgres.PostgresNetty;
+import io.crate.protocols.ssl.SslSettings;
+import io.crate.replication.logical.LogicalReplicationSettings;
+import io.crate.testing.Asserts;
+import io.crate.testing.SQLErrorMatcher;
+import io.crate.testing.SQLTransportExecutor;
+import io.crate.testing.TestingHelpers;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+
+public class PgTunnelLogicalReplicationITest extends ESTestCase {
```

Comment:
Whats the reason to not extend it from `LogicalReplicationTestCase`?

Reply:
Mostly because I wanted more control over the cluster creation due to the different settings used (hba and ssl).

I could try to follow up on this to share some more logic